packages/dapp/src/components/Listings.tsx

```diff
@@ -39,19 +40,29 @@ class Listings extends React.Component<{}, ListingsState> {
     return (
       <StyledDiv>
         whaddup listings: {this.state.applications.toString()}
+        <br/>
+        {this.state.error && this.state.error}
       </StyledDiv>
     );
   }
 
-  private initListings = () => {
-
+  private initListings = async () => {
     const civil = new Civil();
-    const tcr = civil.tcrSingletonTrusted();
-    const instance = this;
-    const subscription = tcr.listingsInApplicationStage().subscribe((listing) => {
-      instance.setState({applications: this.state.applications.push(listing)});
-    });
-    this.setState({applicationSubscription : subscription});
+    let tcr;
+    try {
+      tcr = civil.tcrSingletonTrusted();
+    } catch (ex) {
+      console.log("failed to get tcr.");
+      this.setState({error: "No Supported Network Found. Please set MetaMask network to Rinkeby and Unlock Account."});
+    }
+
+    if (tcr) {
+      const instance = this;
```

Comment:
you don't need this with => functions. The arrow means the scope is the same as the outer function